packages/headless/src/controllers/quickview/case-assist-headless-quickview.test.ts

```diff
@@ -1,170 +1,133 @@
+    );
   });
 
-  it('it adds the correct reducers to engine', () => {
+  it('it adds the case-assist specific reducers to engine', () => {
     expect(engine.addReducers).toHaveBeenNthCalledWith(1, {
       documentSuggestion,
     });
-
-    expect(engine.addReducers).toHaveBeenNthCalledWith(2, {
-      configuration,
-      resultPreview,
-    });
-  });
-
-  it('exposes a subscribe method', () => {
-    expect(quickview.subscribe).toBeTruthy();
   });
 
-  describe('#fetchResultContent', () => {
-    const uniqueId = '1';
-    const requestedOutputSize = 0;
-
-    beforeEach(() => {
-      options.result = buildMockResult({uniqueId});
-      initQuickview();
-
-      quickview.fetchResultContent();
-    });
-
-    it('dispatches a #fetchResultContent action with the result uniqueId', () => {
-      const action = engine.findAsyncAction(fetchResultContent.pending);
-      expect(action?.meta.arg).toEqual({uniqueId, requestedOutputSize});
-    });
+  it('dispatches a preparePreviewPagination action on initialization', () => {
+    const mockedPreparePreviewPagination = jest.mocked(
+      preparePreviewPagination
+    );
 
-    it('dispatches a quickview document suggestion click event', () => {
-      const result = buildMockResult();
-      const thunk = logQuickviewDocumentSuggestionClick(result.uniqueId);
-      const action = engine.findAsyncAction(thunk.pending);
-      expect(action).toBeTruthy();
+    expect(mockedPreparePreviewPagination).toHaveBeenCalledWith({
+      results: engineState.documentSuggestion.documents,
     });
+    expect(engine.dispatch).toHaveBeenCalledWith(
+      mockedPreparePreviewPagination.mock.results[0].value
+    );
   });
 
-  it(`when configured result uniqueId matches the uniqueId in state,
-  #state.content returns the content in state`, () => {
-    const uniqueId = '1';
-    const content = '<div></div>';
-
-    engine.state.resultPreview = buildMockResultPreviewState({
-      uniqueId,
-      content,
-    });
-    options.result = buildMockResult({uniqueId});
-    initQuickview();
-
-    expect(quickview.state.content).toEqual(content);
-  });
+  describe('state', () => {
+    it('exposes the core quickview state', () => {
+      coreQuickviewState.content = '🥔';
 
-  it(`when configured result uniqueId matches the uniqueId in state,
-  #state.content returns an empty string`, () => {
-    engine.state.resultPreview = buildMockResultPreviewState({
-      uniqueId: '1',
-      content: '<div></div>',
+      expect(quickview.state.content).toBe('🥔');
     });
-    options.result = buildMockResult({uniqueId: '2'});
-    initQuickview();
-
-    expect(quickview.state.content).toEqual('');
-  });
 
-  [true, false].forEach((testValue) => {
-    it(`when the result #hasHtmlVersion is ${testValue} #state.resultHasPreview should be ${testValue}`, () => {
-      options.result = buildMockResult({hasHtmlVersion: testValue});
-      initQuickview();
+    it('exposes `currentDocument` in the state ', () => {
```

Comment:
In the case of those "derived" controller state properties, I think we should fully test the intended behavior, because the implementation is in the controller code.

In this specific case, we expect state.currentDocument to return the 1-based position of the current document suggestion.

This is already what we're testing, but I think the test description should be more specific, e.g.:

'#state.currentDocument returns the 1-based position of the document suggestion being quick viewed'

Reply:
The fact that we're testing potentially complex behaviors in controller state getter properties makes me think that this kind of stuff "smells" a little bit. Maybe it would be preferable, in general, to test those selectors separately, and then test that a derived state property is calling the correct selector...

On the other hand, this might be introducing a new layer of abstraction just for the sake of it, so it's probably not worth the effort. It's just ever so slightly annoying me 😅

Reply:
> The fact that we're testing potentially complex behaviors in controller state getter properties makes me think that this kind of stuff "smells" a little bit. Maybe it would be preferable, in general, to test those selectors separately, and then test that a derived state property is calling the correct selector...
>
>On the other hand, this might be introducing a new layer of abstraction just for the sake of it, so it's probably not worth the effort. It's just ever so slightly annoying me 😅

You are correct on both accounts I think:
 - Yes, complex 'state driller' should probably be isolated in selectors, and tested on their own.
 - Yes, doing so is probably ain't worth the hassle, especially on 'old static' code.
 
 However, we should I think keep that in mind for the future ;)
 
 In the meantime, for there, let's test it in the controller.